superset/views/filters.py

```diff
@@ -0,0 +1,44 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from flask_appbuilder.models.filters import BaseFilter
+from flask_babel import lazy_gettext
+
+from superset import security_manager
+
+# pylint: disable=too-few-public-methods
+
+
+class FilterRelatedOwners(BaseFilter):
+    """
+    A filter to allow searching for related owners of a resource.
+
+    Use in the api by adding something like:
+    related_field_filters = {
+      "owners": RelatedFieldFilter("first_name", FilterRelatedOwners),
+    }
+    """
+
+    name = lazy_gettext("Owner")
+    arg_name = "owners"
+
+    def apply(self, query, value):
+        user_model = security_manager.user_model
+        like_value = "%" + value + "%"
+        return query.filter(
+            # could be made to handle spaces between names more gracefully
+            (user_model.first_name + " " + user_model.last_name).ilike(like_value)
```

Comment:
should this `or` a bunch of filters together? I'd imagine this should support searching within a username string too.

Reply:
I originally had it `or` the full name and the username, but @dpgaspar gave feedback that it was strange to get results by username when username is not displayed in the UI, so I removed that.

Using an `or` on the first name and last name would have the effect that you cannot search by full name, so appending them first is preferable. But ultimately a full text search solution would probably be best. These are product/design questions, so I'd prefer to merge this the way it is and iterate on it.

Reply:
I agree that appending (while not strictly correct) is an ok way to handle first names and last names. but I still think it makes sense to include the username `or` filter, even if it might not be clear why they were included. Returning too many results is better than not returning the one you're looking for

Or perhaps the select box should include both the first name last name combo and the username in parens afterwards. I think Jira uses a similar pattern so that you can search both by name and email when tagging someone. Github also does a similar thing with name/username